pinot-common/src/main/java/org/apache/pinot/common/metadata/segment/SegmentPartitionMetadata.java

```diff
@@ -48,6 +53,21 @@ public SegmentPartitionMetadata(
       @Nonnull @JsonProperty("columnPartitionMap") Map<String, ColumnPartitionMetadata> columnPartitionMap) {
     Preconditions.checkNotNull(columnPartitionMap);
     _columnPartitionMap = columnPartitionMap;
+    _uploadedSegmentPartitionId = -1;
```

Comment:
Good to define a constant that can be used at other places directly checking for `-1`, may be with a better name.

```
public static final int NON_EXTERNAL_PARTITION_ID = -1;
```